creator-node/src/snapbackSM/secondarySyncHealthTracker.js

```diff
@@ -150,14 +66,119 @@ const SecondarySyncHealthTracker = {
    */
-  _getRedisKeyPattern (secondary, wallet, syncType, outcome, date = null) {
+  _getRedisKeyPattern ({ secondary = '*', wallet = '*', syncType = '*', outcome = '*', date = null }) {
     const prefix = 'SecondarySyncRequestOutcomes-Daily'
 
     // format: YYYY-MM-DD
     date = date || new Date().toISOString().split('T')[0]
 
     return `${prefix}:::${secondary}:::${wallet}:::${syncType}:::${date}:::${outcome}`
+  },
+
+  _parseSecondaryFromRedisKey (key) {
+    return key.split(':::')[1]
+  }
+}
+
+const Setters = {
+  async recordSuccess (secondary, wallet, syncType) {
+    await Setters._recordSyncRequestOutcome(secondary, wallet, syncType, true)
+  },
+
+  async recordFailure (secondary, wallet, syncType) {
+    await Setters._recordSyncRequestOutcome(secondary, wallet, syncType, false)
+  },
+
+  async _recordSyncRequestOutcome (secondary, wallet, syncType, success = true) {
+    try {
+      const outcome = success ? Outcomes.SUCCESS : Outcomes.FAILURE
+      const redisKey = Utils._getRedisKeyPattern({ secondary, wallet, syncType, outcome })
+
+      // incr() will create key with value 0 if non-existent
+      await redisClient.incr(redisKey)
+
+      // Set key expiration time (sec) in case it hasn't already been set (prob not most efficient)
+      await redisClient.expire(redisKey, DailyRedisKeyExpirationSec)
+
+      logger.info(`SecondarySyncHealthTracker:_recordSyncRequestOutcome || Recorded ${redisKey}`)
+
+      // Swallow error + log
+    } catch (e) {
+      logger.error(`SecondarySyncHealthTracker:_recordSyncRequestOutcome Error || ${e.message}`)
+    }
+  }
+}
+
+const Getters = {
+  /**
+   * Given wallet and secondaries array, returns map from each secondary to SuccessCount, FailureCount, and SuccessRate
+   *
+   * @param {String} wallet
+   * @param {Array} secondaries
+   * @returns {Object} { secondary1: { 'SuccessCount' : _, 'FailureCount': _, 'SuccessRate': _ }, ... }
+   */
+  async computeUserSecondarySyncSuccessRates (wallet, secondaries) {
+    // Initialize sync success and failure counts for every secondary to 0
+    let secondarySyncMetrics = {}
+    secondaries.forEach(secondary => {
+      secondarySyncMetrics[secondary] = { 'SuccessCount': 0, 'FailureCount': 0 }
+    })
+
+    // Retrieve map of all SyncRequestOutcome keys and daily counts for user from all secondaries
+    const userSecondarySyncHealthOutcomes = await Getters.getSyncRequestOutcomeMetrics({ wallet })
+
+    // Aggregate all daily SyncRequest outcome counts by secondary
+    for (let [key, count] of Object.entries(userSecondarySyncHealthOutcomes)) {
+      count = parseInt(count)
+      const secondary = Utils._parseSecondaryFromRedisKey(key)
+
+      if (!(secondary in secondarySyncMetrics)) {
+        // This case can be hit for old secondaries that have been cycled out of user's replica set - these can be safely skipped
+        continue
+      }
+
+      if (key.includes('Success')) {
```

Comment:
this could call the refactored version of _parseSecondaryFromRedisKey so as to not parse the string of the key

Reply:
beaut, done